libraries/hwlib/library/targets/usart_queue.hpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * =====================================================================================
+ *
+ *       Filename:  usart_queue.hpp
+ *
+ *    Description:  
+ *
+ *        Version:  1.0
+ *        Created:  14-03-19 11:25:21
+ *       Revision:  none
+ *       Compiler:  gcc
+ *
+ *         Author:  YOUR NAME (), 
+ *   Organization:  
+ *
+ * =====================================================================================
+ */
+#ifndef QUEUE_HPP
+#define QUEUE_HPP
+#include "hwlib.hpp"
+
+template <class T, size_t QUEUE_SIZE>
+class Queue {
+private:
+	unsigned int _front, _back, _count;
+	T _data[QUEUE_SIZE];
+
+public:
+	Queue() {
+		_front = 0;
+		_back = 0;
+		_count = 0;
+	}
+	inline int count();
```

Comment:
Why do these functions return a int? the variable is a unsigned int at the moment.